enhancements/workload-partitioning/rationalizing-configuration.md

```diff
@@ -0,0 +1,478 @@
+3. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+#### Day 2: Add a New MachineConfigPool
+
+Adding a new MachineConfigPool is safe only if the MachineConfigPool
+has a PerformanceProfile associated and the PAO is installed.
+
+1. PAO reads PerformanceProfile to generate new MachineConfig for the
+   pool with CRI-O and `kubelet` settings.
+2. MachineConfigs are applied to appropriate nodes, rebooting them in
+   the process.
+3. Kubelet starts and sees the config file enabling partitioning. It
+   advertises the workload resource type on the Node and mutates
+   static pods with partitioning annotations.
+4. CRI-O sees pods with workload annotations and uses the resource
+   request to set cpushares and cpuset.
+
+If the MachineConfigPool does not match a PerformanceProfile, there
+will be no cpuset information and the PAO will generate MachineConfigs
+with partitioning enabled but not tied to a cpuset. See the [Risks
+section](#risks-and-mitigations) for details.
+
+If the PAO is not installed, it will not generate the overriding
+MachineConfigs for the new MachineConfigPool. See the [Risks
+section](#risks-and-mitigations) for details.
+
+### Risks and Mitigations
+
+There is some risk of shipping a feature with part of the
+implementation in the OpenShift release payload but the enabling tool
+delivered separately. PAO is considered to be a "core" OpenShift
+component, even though it is delivered separately. There is not a
+separate SKU for it, for example. The PAO team has been working
+closely with the node team on the implementation of this feature, so
+we do not anticipate any issues delivering the finished work in this
+way.
+
+If a MachineConfigPool exists without a matching PerformanceProfile,
+there will be no cpuset information and the PAO will generate
+MachineConfigs with partitioning enabled but not tied to a cpuset.
+This is somewhat safe, but may lead to unexpected behavior. The
+mutated pods would float across the full cpuset in the same way as if
+partitioning was not enabled and the cpushares would not be deducted
+from available CPUs, potentially leading to over commit scenarios.
+
+If the PAO is not installed into a cluster with workload partitioning
+enabled and configured, adding a new MachineConfigPool can be unsafe.
+The nodes in the pool will not enable partitioning at all. This may
+not be safe, since kubelet will not advertise the workload resource
+type but the admission plugin will mutate pods to require it. The
+scheduler may refuse to place management workloads on the nodes in the
+pool.
+
+## Design Details
+
+### Open Questions
+
+1. What runs the PAO `render` command in the assisted installer
+   workflow?
+2. How hard do we need to work to ensure that the PAO version matches
+   the release payload version for the cluster being deployed?
+   What/who is responsible for that?
+3. We need to ensure MCO render handles the KubeletConfig CR generated
```

Comment:
I guess some of this wasnt' clear to me but mostly what we're talking about is PAO->kubelet/containerruntimes CRDs? or is PAO looking to create its own that it's expecting the MCO to render into a renderedconfig?

Reply:
The PAO does define its own PerformanceProfile CRD, and it renders them down to KubeletConfig and MachineConfig resources already. Some of the work for this release will be to extend that to include the new settings, but we don't expect the MCO to understand any new APIs.

https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md#high-level-end-to-end-workflow has a bit more detail about what's being done manually today that we're trying to automate here.